cmd/kube-apiserver/apiserver.go

```diff
@@ -34,6 +35,13 @@ import (
 )
 
 func main() {
+	if err := runAPIServerCmd(); err != nil {
```

Comment:
This should be defer'd. 
This code will still have issues is something does an Exit.
It will also fail to log anything if an exception is thrown.